stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/header/MessageListHeaderViewStyle.kt

```diff
@@ -0,0 +1,174 @@
+package io.getstream.chat.android.ui.message.list.header
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import androidx.annotation.ColorInt
+import androidx.core.content.ContextCompat
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.TransformStyle
+import io.getstream.chat.android.ui.common.extensions.internal.getColorCompat
+import io.getstream.chat.android.ui.common.extensions.internal.getDimension
+import io.getstream.chat.android.ui.common.extensions.internal.getDrawableCompat
+import io.getstream.chat.android.ui.common.extensions.internal.use
+import io.getstream.chat.android.ui.common.style.TextStyle
+
+/**
+ * Style for [MessageListHeaderView].
+ * Use this class together with [TransformStyle.messageListHeaderStyleTransformer] to change [MessageListHeaderView] styles programmatically.
+ *
+ * @property titleTextStyle - appearance for title text
+ * @property offlineTextStyle - appearance for offline text
+ * @property searchingForNetworkTextStyle - appearance for searching for network text
+ * @property onlineTextStyle - appearance for online text
+ * @property showUserAvatar - shows/hides user avatar. Shown by default
+ * @property backButtonIcon - icon for back button. Default - [R.drawable.stream_ui_arrow_left]
+ * @property showBackButton - shows/hides back button. Shown by default
+ * @property showBackButtonBadge - shows/hides unread badge. Hidden by default
+ * @property backButtonBadgeBackgroundColor - unread badge color. Default - [R.color.stream_ui_accent_red]
+ * @property showSearchingForNetworkProgressBar - shows/hides searching for network progress bar. Shown by default
+ * @property searchingForNetworkProgressBarTint - progress bar tint color. Default - [R.color.stream_ui_accent_blue]
+ */
+public data class MessageListHeaderViewStyle(
+    public val titleTextStyle: TextStyle,
+    public val offlineTextStyle: TextStyle,
+    public val searchingForNetworkTextStyle: TextStyle,
+    public val onlineTextStyle: TextStyle,
+    public val showUserAvatar: Boolean,
+    public val backButtonIcon: Drawable,
+    public val showBackButton: Boolean,
+    public val showBackButtonBadge: Boolean,
+    @ColorInt public val backButtonBadgeBackgroundColor: Int,
+    public val showSearchingForNetworkProgressBar: Boolean,
+    public val searchingForNetworkProgressBarTint: ColorStateList,
+) {
+
+    internal companion object {
+        operator fun invoke(context: Context, attrs: AttributeSet?): MessageListHeaderViewStyle {
+            context.obtainStyledAttributes(
```

Comment:
Obtain individual attrs listed in `MessageListHeaderView` from view layout attrs, or from `streamUiMessageListHeaderStyle` style in our theme

Reply:
Wow, it looks like black magic. If everything works so smooth there is no problem between XML overriding for theme/style

Reply:
Yup, that's what I needed 🥇